src/iotivity/iotivity_tools.cc

```diff
@@ -0,0 +1,239 @@
+** http://www.apache.org/licenses/LICENSE-2.0
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+** $CISCO_END_LICENSE$
+**
+****************************************************************************/
+#include "iotivity/iotivity_tools.h"
+
+char *pDebugEnv = NULL;
+
+void PrintfOcResource(const OCResource & oCResource) {
+    DEBUG_MSG("PrintfOcResource\n");
+    DEBUG_MSG("Res[sId] = %s\n", oCResource.sid().c_str());
+    DEBUG_MSG("Res[Uri] = %s\n", oCResource.uri().c_str());
+    DEBUG_MSG("Res[Host] = %s\n", oCResource.host().c_str());
+    DEBUG_MSG("Res[Resource types] \n");
+
+    for (auto &resourceTypes : oCResource.getResourceTypes()) {
+        DEBUG_MSG("\t\t%s\n", resourceTypes.c_str());
+    }
+
+    DEBUG_MSG("Res[Resource interfaces] \n");
+
+    for (auto &resourceInterfaces : oCResource.getResourceInterfaces()) {
+        DEBUG_MSG("\t\t%s\n", resourceInterfaces.c_str());
+    }
+}
+
+void PrintfOcRepresentation(const OCRepresentation & oCRepr) {
+    DEBUG_MSG("PrintfOcRepresentation\n");
+
+    for (auto& cur : oCRepr) {
+        std::string attrname = cur.attrname();
+
+        if (AttributeType::String == cur.type()) {
+            std::string curStr = cur.getValue<string>();
+            DEBUG_MSG("Rep[String]: key=%s, value=%s\n",
+                      attrname.c_str(), curStr.c_str());
+        } else if (AttributeType::Integer == cur.type()) {
+            DEBUG_MSG("Rep[Integer]: key=%s, value=%d\n",
+                      attrname.c_str(), cur.getValue<int>());
+        } else if (AttributeType::Double == cur.type()) {
+            DEBUG_MSG("Rep[Double]: key=%s, value=%f\n",
+                      attrname.c_str(), cur.getValue<double>());
+        } else if (AttributeType::Boolean == cur.type()) {
+            DEBUG_MSG("Rep[Boolean]: key=%s, value=%d\n",
+                      attrname.c_str(), cur.getValue<bool>());
+        }
+    }
+}
+
+static void UpdateDestOcRepresentationString(OCRepresentation & oCReprDest,
+                                             std::string attributeName,
+                                             std::string value) {
+    for (auto& cur : oCReprDest) {
+        std::string attrname = cur.attrname();
+        if (attrname == attributeName) {
+            if (AttributeType::String == cur.type()) {
+                std::string oldValue = cur.getValue<string>();
+                cur = value;
+                DEBUG_MSG("Updated[%s] old=%s, new=%s\n",
+                          attrname.c_str(), oldValue.c_str(), value.c_str());
+            } else {
+                DEBUG_MSG("Updated[%s] mismatch String\n", attrname.c_str());
+                return;
+            }
+        }
```

Comment:
It iterates over all representation item because there is no OCRepresentation.find() from iterator.
This function only updates the "attributename" representation (once done, it returns, no further iteration)